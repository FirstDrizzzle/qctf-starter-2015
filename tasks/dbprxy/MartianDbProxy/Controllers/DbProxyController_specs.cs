using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalTests.Wpf;
using Language;
using NUnit.Framework;

namespace MartianDbProxy.Controllers
{
	[TestFixture]
	[UseReporter(typeof(DiffReporter))]
	public class DbProxyController_specs
	{
	    private DbProxyController controller;


	    [SetUp]
	    public void SetUp()
	    {
            controller = new DbProxyController();
	    }

		[Test]
		public void greatings_on_get()
		{
			var response = controller.Post(null);
			Approvals.Verify(response.ErrorMessage.ToEarth() + "\n");
		}

		[Test]
		public void access_to_all_content()
		{
			var responses = controller.Contents.Select(c => controller.Post(new AccessRequest { SecurityToken = c.SecurityToken.ToMars(), ApiKey = DbProxyController.apiKey.ToMars()})).ToList();
		    foreach (var response in responses)
                Assert.IsNull(response.ErrorMessage);
            Approvals.VerifyAll(responses.Select(r => r.NextSecurityLevelChallenge), "responseNextChallenge");
		}
        
        [Test]
		public void no_acces_if_wrong_securityToken()
		{
            var response = controller.Post(new AccessRequest { SecurityToken = "sdfasfdasfdasdfasdf", ApiKey = DbProxyController.apiKey.ToMars() });
			Approvals.Verify(response.ErrorMessage.ToEarth());
		}
		[Test]
		public void no_acces_if_wrong_apiKey()
		{
            var response = controller.Post(new AccessRequest { SecurityToken = controller.Contents[0].SecurityToken, ApiKey = "badKey" });
			Approvals.Verify(response.ErrorMessage.ToEarth());
		}

		[Test]
		public void no_acces_if_no_apiKey()
		{
            var response = controller.Post(new AccessRequest { SecurityToken = controller.Contents[0].SecurityToken });
			Approvals.Verify(response.ErrorMessage.ToEarth());
		}
	}
}