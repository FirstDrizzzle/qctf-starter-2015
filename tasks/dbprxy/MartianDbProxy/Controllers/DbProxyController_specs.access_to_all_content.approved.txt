responseNextChallenge[0] = AAF893E1D631430799F7F0FF27308162
responseNextChallenge[1] = 2DD4F4C12C3A43BDA3AD5F707725DA74
responseNextChallenge[2] = BC7017B8A3E4466D98B80F762D47CB15
responseNextChallenge[3] = 
